### PR TITLE
Merge rc v1.4.0 in multisigner

### PR DESCRIPTION
**Pull request by iulianpascalau**
## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- merge
  
## Proposed Changes
- merge

## Testing procedure
- N/A

### CODE REVIEW

**Review by sstanculeanu — DISMISSED** [review]

File: consensus/spos/bls/subroundSignature.go [hunk 1]
```diff
@@ -368,7 +363,19 @@ func (sr *subroundSignature) doSignatureJobForManagedKeys() bool {
...
 
 		managedPrivateKey := sr.GetMessageSigningPrivateKey(pkBytes)
-		signatureShare, err := sr.MultiSigner().CreateAndAddSignatureShareForKey(sr.GetData(), managedPrivateKey, pkBytes)
+		selfIndex, err := sr.ConsensusGroupIndex(pk)
+		if err != nil {
+			log.Warn("doSignatureJobForManagedKeys: index not found", "pk", pkBytes)
+			continue
+		}
+
+		managedPrivateKeyBytes, err := managedPrivateKey.ToByteArray()
```
Comment (AdoAdoAdo) [inline consensus/spos/bls/subroundSignature.go:372]: signature handler was supposed to do itself the management of private keys and signing and move that out of the consensus, but here it is partially moved into consensus again.
Maybe add a todo and I will do the refactor, or you can move the management into the signature handler.

File: consensus/interface.go [hunk 2]
```diff
@@ -153,6 +153,7 @@ type ScheduledProcessor interface {
 type SignatureHandler interface {
 	Reset(pubKeys []string) error
 	CreateSignatureShare(msg []byte, index uint16, epoch uint32) ([]byte, error)
+	CreateSignatureShareWithPrivateKey(message []byte, index uint16, epoch uint32, privateKeyBytes []byte) ([]byte, error)
```
Comment (AdoAdoAdo) [inline consensus/interface.go:156]: this needs to be refactored to remove private key as parameter.
If you add a TODO I can do the refactor.

File: consensus/spos/bls/subroundSignature.go (hunk 1 above)
Comment (iulianpascalau) [reply consensus/spos/bls/subroundSignature.go:373]: added TODO

File: consensus/interface.go (hunk 2 above)
Comment (iulianpascalau) [reply consensus/interface.go:156]: added TODO

**Review by AdoAdoAdo — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Pull request merged**